visidata/loaders/_pandas.py

```diff
@@ -75,6 +79,34 @@ def sort(self):
             ascending += [not reverse] * len(cols)
         self.rows.sort_values(by=by_cols, ascending=ascending, inplace=True)
 
+    def _checkSelectedIndex(self):
+        if self._selectedMask.index is not self.df.index:
+            # self.df was modified inplace, so the selection
+            # is no longer valid. Likely to heavy but erring
+            # on side of correctness here.
+            self._selectedMask = pd.Series(False, index=self.df.index)
+
+    # TODO: add vectorized implementation for multiple rows
+    def isSelected(self, row):
+        if row is None:
+            return False
+        self._checkSelectedIndex()
+        return self._selectedMask.loc[row.name]
+    def selectRow(self, row):
+        'Select given row. O(log n)'
+        self._checkSelectedIndex()
+        self._selectedMask.loc[row.name] = True
+    def unselectRow(self, row):
+        self._checkSelectedIndex()
+        self._selectedMask.loc[row.name] = False
+    @property
+    def nSelected(self):
+        self._checkSelectedIndex()
+        return self._selectedMask.sum()
+    @property
+    def selectedRows(self):
+        self._checkSelectedIndex()
+        return self.df.loc[self._selectedMask]
```

Comment:
`selectedRows` is probably wrong here, the result doesn't have the right iterable output. Probably need to wrap in a DataFrameAdapter

Reply:
ah yeah, or a different kind of adapter.  One thing to keep in mind, is that selectedRows should return a list of rows in the same order as the sheet itself.  This is for things like `dup-sheet` (`"`) to feel right. I think your approach already does this, but I wanted to make sure it's explicit.